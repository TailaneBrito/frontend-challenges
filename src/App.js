import "./App.css";
import Board from "./components/Board";
import ChessStyleOnly from "./components/ChessStyleOnly";
import { DetailsBody, Details } from "./styles/board.style";

function App() {
  return (
    <div className="App">
      <div
        style={{
          padding: "32px",
          display: "flex",
          flexDirection: "row",
          gap: "32px",
          justifyContent: "center",
          flexWrap: "wrap",
        }}
      >
        <Details open>
          <summary>Board v1</summary>
          <DetailsBody>
            <Board />
          </DetailsBody>
        </Details>

        <Details open>
          <summary>Board v2</summary>
          <DetailsBody>
            <ChessStyleOnly />
          </DetailsBody>
        </Details>
      </div>
    </div>
  );
}

export default App;
