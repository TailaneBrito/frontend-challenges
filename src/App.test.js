import { render, screen } from "@testing-library/react";
import App from "./App";

describe("App", () => {
  it("renders learn react link", () => {
    render(<App />);
    const linkElement = screen.getByText(/Board v1/i);
    expect(linkElement).toBeInTheDocument();
    const linkElement2 = screen.getByText(/Board v2/i);
    expect(linkElement2).toBeInTheDocument();
  });

  it("renders correctly again", () => {
    render(<App />);
  });
});
