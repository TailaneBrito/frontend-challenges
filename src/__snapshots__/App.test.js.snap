// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <div
    class="App"
  >
    <div
      style="display: flex; flex-direction: row; gap: 32px; justify-content: center; flex-wrap: wrap;"
    >
      <details>
        <summary>
          Board v1
        </summary>
        <div
          style="display: flex; align-content: center; justify-content: center; align-items: center; margin: 0px auto; background: orange; width: auto;"
        >
          <div
            class="sc-aYaIB ftDnVu"
          >
            <div
              class="sc-fqkwJk gabvji"
            >
              <div
                class="sc-dcJtft ciJpPp"
              >
                A
              </div>
              <div
                class="sc-dcJtft ciJpPp"
              >
                B
              </div>
              <div
                class="sc-dcJtft ciJpPp"
              >
                C
              </div>
              <div
                class="sc-dcJtft ciJpPp"
              >
                D
              </div>
              <div
                class="sc-dcJtft ciJpPp"
              >
                E
              </div>
              <div
                class="sc-dcJtft ciJpPp"
              >
                F
              </div>
              <div
                class="sc-dcJtft ciJpPp"
              >
                G
              </div>
              <div
                class="sc-dcJtft ciJpPp"
              >
                H
              </div>
            </div>
            <div
              style="display: flex; flex-direction: column;"
            >
              <div
                class="sc-eqUzNf jKYEVq"
              >
                8
              </div>
              <div
                class="sc-eqUzNf jKYEVq"
              >
                7
              </div>
              <div
                class="sc-eqUzNf jKYEVq"
              >
                6
              </div>
              <div
                class="sc-eqUzNf jKYEVq"
              >
                5
              </div>
              <div
                class="sc-eqUzNf jKYEVq"
              >
                4
              </div>
              <div
                class="sc-eqUzNf jKYEVq"
              >
                3
              </div>
              <div
                class="sc-eqUzNf jKYEVq"
              >
                2
              </div>
              <div
                class="sc-eqUzNf jKYEVq"
              >
                1
              </div>
            </div>
            <div
              class="sc-iGgVNO fMAPrq"
            >
              <div
                class="sc-gsFSjX eOBvup"
              >
                <div
                  class="sc-kAycRU fBVwSz"
                >
                  <div
                    class="sc-dhKdPU jZTktB"
                  >
                    <div
                      class="sc-kpDprT bVUMVx"
                    >
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                    </div>
                  </div>
                  <div
                    class="sc-dhKdPU jZTktB"
                  >
                    <div
                      class="sc-kpDprT bVUMVx"
                    >
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                    </div>
                  </div>
                  <div
                    class="sc-dhKdPU jZTktB"
                  >
                    <div
                      class="sc-kpDprT bVUMVx"
                    >
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                    </div>
                  </div>
                  <div
                    class="sc-dhKdPU jZTktB"
                  >
                    <div
                      class="sc-kpDprT bVUMVx"
                    >
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                    </div>
                  </div>
                  <div
                    class="sc-dhKdPU jZTktB"
                  >
                    <div
                      class="sc-kpDprT bVUMVx"
                    >
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                    </div>
                  </div>
                  <div
                    class="sc-dhKdPU jZTktB"
                  >
                    <div
                      class="sc-kpDprT bVUMVx"
                    >
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                    </div>
                  </div>
                  <div
                    class="sc-dhKdPU jZTktB"
                  >
                    <div
                      class="sc-kpDprT bVUMVx"
                    >
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                    </div>
                  </div>
                  <div
                    class="sc-dhKdPU jZTktB"
                  >
                    <div
                      class="sc-kpDprT bVUMVx"
                    >
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                      <div
                        class="sc-jXbVAB dcQoiq"
                      />
                      <div
                        class="sc-imWZod ATHNR"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </details>
      <details>
        <summary>
          Board v2
        </summary>
        <div
          class="chessboard"
          style="display: flex; width: 400px; border: 1px solid orange;"
        >
          <div
            style="display: flex; flex-direction: column;"
          >
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
          </div>
          <div
            style="display: flex; flex-direction: column;"
          >
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
          </div>
          <div
            style="display: flex; flex-direction: column;"
          >
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
          </div>
          <div
            style="display: flex; flex-direction: column;"
          >
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
          </div>
          <div
            style="display: flex; flex-direction: column;"
          >
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
          </div>
          <div
            style="display: flex; flex-direction: column;"
          >
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
          </div>
          <div
            style="display: flex; flex-direction: column;"
          >
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
          </div>
          <div
            style="display: flex; flex-direction: column;"
          >
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
            <div
              style="background: black; width: 50px; height: 50px;"
            />
            <div
              style="background: white; width: 50px; height: 50px;"
            />
          </div>
        </div>
      </details>
    </div>
  </div>
</div>
`;
